### DISCUSSION TITLE
Change request for hover time over preview window and hover time over pinned icons (with no open windows)

### DISCUSSION THREAD

**Discussion by favoxhille** [HerMajestyDrMona/Windows11DragAndDropToTaskbarFix#61] [Ideas]
Hi,
First of all I want to grantulate with the creator for creating a very useful script and the community for suggesting ways of improving it.

So my 2 cents on 2 configure paramaters that could be added to make this script even better

1.  Adding a paramater for wait time to hover over window preview before opening it, because we can change the time it takes to automatically open when setting AutoOpenFirstWindowInBestMethodEver=2 or 1 but not the time when we set it to 0 or to 2 and have multiple windows open
2. It's nice that we can open pinned apps but there's no parameter to change the wait time before the app opens, actually a better solution ,if possible, would be to open the pinned app only when mouse 1 is released so to avoid accidentely opening an app that we don't intend to use

**Comment by HerMajestyDrMona** [comment]
Hi. Thanks for your feedback.

> Adding a paramater for wait time to hover over window preview before opening it, because we can change the time it takes to automatically open when setting AutoOpenFirstWindowInBestMethodEver=2 or 1 but not the time when we set it to 0 or to 2 and have multiple windows open

From what I remember it can not be done. When `AutoOpenFirstWindowInBestMethodEver=2` and when multiple windows are open - it just displays the preview windows. This is Windows behavior, so the time after moving the mouse to these preview windows is depended on Windows, not this program. There might be some registry keys to configure this time, but I am not familiar with them. So in short: the preview windows are displayed by Windows 11 and also the behavior after moving the mouse pointer with the dragged file to them is controlled by Windows, not this program.
You can however configure `HowLongKeepMouseOverAppIconBeforeAutoOpeningMilliseconds=550` to increase the time before auto-opening windows, when `AutoOpenFirstWindowInBestMethodEver=1` or `2`.

> It's nice that we can open pinned apps but there's no parameter to change the wait time before the app opens, actually a better solution ,if possible, would be to open the pinned app only when mouse 1 is released so to avoid accidentely opening an app that we don't intend to use

This also should depend on `HowLongKeepMouseOverAppIconBeforeAutoOpeningMilliseconds=550`. 

https://github.com/HerMajestyDrMona/Windows11DragAndDropToTaskbarFix/blob/8162d548d66ed8b35ef005023dc80c24853a5cc7/Windows11DragAndDropToTaskbarFix/Windows11DragAndDropToTaskbarFix.cpp#L1114-L1134

The mouse button can not be released, because when the file is being dragged it would lose it and require to drag again.
